libraries/EEPROM/EEPROM.h

```diff
@@ -53,8 +53,11 @@ class EEPROMClass {
     if (address < 0 || address + sizeof(T) > _size)
       return t;
 
-    memcpy(_data + address, (const uint8_t*) &t, sizeof(T));
-    _dirty = true;
+    _dirty = (memcmp(_data + address, (const uint8_t*) &t, sizeof(T)) !=0);
```

Comment:
I just changed the lines above. 
put and get are function template and are part of class definition.